src/fields/fieldp128/mod.rs

```diff
@@ -0,0 +1,244 @@
+use std::{
+    cmp::Ordering,
+    fmt::{self, Debug},
+    io::{self, Read},
+    ops::{Add, AddAssign, Mul, Neg},
+};
+
+use anyhow::{Context, anyhow};
+use subtle::ConstantTimeEq;
+
+use crate::{
+    Codec,
+    fields::{
+        FieldElement,
+        fieldp128::ops::{
+            fiat_p128_add, fiat_p128_from_bytes, fiat_p128_from_montgomery,
+            fiat_p128_montgomery_domain_field_element, fiat_p128_mul,
+            fiat_p128_non_montgomery_domain_field_element, fiat_p128_opp, fiat_p128_to_bytes,
+            fiat_p128_to_montgomery,
+        },
+    },
+};
+
+/// FieldP128 is the field with modulus 2^128 - 2^108 + 1, described in [Section 7.2 of
+/// draft-google-cfrg-libzk-00][1]. The field does not get a name in the draft, but P128 comes from
+/// the longfellow implementation ([3]).
+///
+/// Field elements are serialized in little-endian form, per [Section 7.2.1 of draft-google-cfrg-libzk-00][2].
+///
+/// [1]: https://www.ietf.org/id/draft-google-cfrg-libzk-00.html#section-7.2
+/// [2]: https://www.ietf.org/id/draft-google-cfrg-libzk-00.html#section-7.2.1
```

Comment:
Let's update these links to a version that won't disappear once the next draft is published

```suggestion
/// draft-google-cfrg-libzk-01][1]. The field does not get a name in the draft, but P128 comes from
/// the longfellow implementation ([3]).
///
/// Field elements are serialized in little-endian form, per [Section 7.2.1 of draft-google-cfrg-libzk-01][2].
///
/// [1]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-7.2
/// [2]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-7.2.1
```